h2o-algos/src/main/java/hex/pca/mtj/PCA_MTJ_EVD_DenseMatrix.java

```diff
@@ -0,0 +1,54 @@
+package hex.pca.mtj;
+
+import hex.pca.PCAInterface;
+import hex.util.EigenPair;
+import hex.util.LinearAlgebraUtils;
+import no.uib.cipr.matrix.DenseMatrix;
+import no.uib.cipr.matrix.Matrix;
+import no.uib.cipr.matrix.NotConvergedException;
+import water.util.ArrayUtils;
+
+/**
+ * @author mathemage <ha@h2o.ai>
+ *         created on 1.5.17
+ */
+public class PCA_MTJ_EVD_DenseMatrix implements PCAInterface {
+  private static DenseMatrix gramMatrix;
```

Comment:
Why are these fields static? It doesn't seem like they should

Reply:
@michalkurka sorry, my bad. Somehow staticity creeped in :-) I have made them non-static. Thank you for pointing it out :+1: :smile: